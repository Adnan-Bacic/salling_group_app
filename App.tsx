/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';
import {
  StatusBar,
  SafeAreaView,
  StyleSheet,
} from 'react-native';
import { Provider as StoreProvider } from 'react-redux';
import store from './src/redux/store';
import Navigator from './src/navigation/Navigator';

const App = (): React.ReactElement => {
  return (
    <>
      <StoreProvider
        store={store}
      >
        <SafeAreaView
          style={styles.safeAreaView}
        >
          <StatusBar />
          <Navigator />
        </SafeAreaView>
      </StoreProvider>
    </>
  );
};

const styles = StyleSheet.create({
  safeAreaView: {
    flex: 1,
  },
});

export default App;
