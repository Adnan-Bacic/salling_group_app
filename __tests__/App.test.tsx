/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

let wrapper: any;

beforeEach(() => {
  wrapper = renderer.create(
    <App />,
  );
})

test('renders correctly', () => {
  const tree = wrapper.toJSON();
  expect(tree).toMatchSnapshot();
});
