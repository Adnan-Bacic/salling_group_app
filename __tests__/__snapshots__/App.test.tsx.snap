// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
          ]
        }
      >
        <View
          accessibilityElementsHidden={false}
          collapsable={false}
          importantForAccessibility="auto"
          pointerEvents="auto"
          removeClippedSubviews={false}
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "flex": 1,
                "top": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <RNSScreenStack
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <RNSScreen
                  collapsable={false}
                  forwardedRef={[Function]}
                  nativeID="animatedComponent"
                  onAppear={[Function]}
                  onDisappear={[Function]}
                  onDismissed={[Function]}
                  onWillDisappear={[Function]}
                  replaceAnimation="pop"
                  stackPresentation="push"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <RNSScreenStackHeaderConfig
                    backButtonInCustomView={false}
                    backgroundColor="rgb(255, 255, 255)"
                    color="rgb(0, 122, 255)"
                    direction="ltr"
                    hidden={false}
                    hideBackButton={false}
                    hideShadow={false}
                    largeTitleHideShadow={false}
                    title="Stack1"
                    titleColor="rgb(28, 28, 30)"
                    topInsetEnabled={false}
                    translucent={false}
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                        },
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View>
                      <Text>
                        stack 1
                      </Text>
                      <View
                        accessibilityRole="button"
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        nativeID="animatedComponent"
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#007AFF",
                                  "fontSize": 18,
                                  "margin": 8,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            nav
                          </Text>
                        </View>
                      </View>
                      <View
                        accessibilityRole="button"
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        nativeID="animatedComponent"
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#007AFF",
                                  "fontSize": 18,
                                  "margin": 8,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            redux
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RNSScreen>
              </RNSScreenStack>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onLayout={[Function]}
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "#ffffff",
            "bottom": 0,
            "elevation": 4,
            "left": 0,
            "position": null,
            "right": 0,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(0, 122, 255)",
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityRole="tablist"
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "marginBottom": 0,
                  "marginHorizontal": 0,
                  "maxWidth": 336,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "selected": true,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "paddingVertical": 6,
                }
              }
            >
              <View
                pointerEvents="none"
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 24,
                      "marginHorizontal": 12,
                      "marginTop": 2,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 0,
                          "position": "absolute",
                          "top": -2,
                        },
                        Object {
                          "right": -2,
                        },
                      ]
                    }
                  >
                    <Text
                      collapsable={false}
                      nativeID="animatedComponent"
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "backgroundColor": "#f50057",
                          "borderRadius": 8,
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontSize": 8,
                          "fontWeight": "400",
                          "height": 16,
                          "lineHeight": 16,
                          "minWidth": 16,
                          "opacity": 0,
                          "overflow": "hidden",
                          "paddingHorizontal": 4,
                          "textAlign": "center",
                          "textAlignVertical": "center",
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  style={
                    Object {
                      "height": 16,
                      "paddingBottom": 2,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                        ]
                      }
                    >
                      StackNavigator1
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "rgba(255, 255, 255, 0.5)",
                            },
                          ],
                        ]
                      }
                    >
                      StackNavigator1
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "selected": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "paddingVertical": 6,
                }
              }
            >
              <View
                pointerEvents="none"
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 24,
                      "marginHorizontal": 12,
                      "marginTop": 2,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "red",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 0,
                          "position": "absolute",
                          "top": -2,
                        },
                        Object {
                          "right": -2,
                        },
                      ]
                    }
                  >
                    <Text
                      collapsable={false}
                      nativeID="animatedComponent"
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "backgroundColor": "#f50057",
                          "borderRadius": 8,
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontSize": 8,
                          "fontWeight": "400",
                          "height": 16,
                          "lineHeight": 16,
                          "minWidth": 16,
                          "opacity": 0,
                          "overflow": "hidden",
                          "paddingHorizontal": 4,
                          "textAlign": "center",
                          "textAlignVertical": "center",
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  style={
                    Object {
                      "height": 16,
                      "paddingBottom": 2,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                        ]
                      }
                    >
                      StackNavigator2
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "rgba(255, 255, 255, 0.5)",
                            },
                          ],
                        ]
                      }
                    >
                      StackNavigator2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</RCTSafeAreaView>
`;
